src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Runtime/PreferStreamAsyncMemoryOverloads.cs

```diff
@@ -230,15 +230,15 @@ private void AnalyzeCompilationStart(CompilationStartAnalysisContext context)
                     {
                         rule = PreferStreamReadAsyncMemoryOverloadsRule;
                         ruleMessageMethod = undesiredReadAsyncMethod.Name;
-                        ruleMessagePreferredMethod = preferredReadAsyncMethod.ToDisplayString(SymbolDisplayFormat.CSharpErrorMessageFormat);
+                        ruleMessagePreferredMethod = string.Join(", ", preferredReadAsyncMethod.GetParameters());
```

Comment:
What does the rendered result message look like now?

Reply:
Before:

> Change the 'ReadAsync' method call to the overload that receives a 'System.IO.Stream.ReadAsync(System.Memory<byte>, System.Threading.CancellationToken)'.

Now:

> Change the 'ReadAsync' method call to the overload that receives a 'System.Memory<byte>, System.Threading.CancellationToken'.

Reply:
Personally, I think it's strange to show the arguments listed like this without the context of the method; that's a bit unusual.  I'd have actually preferred it just updating the message text:
```
Change the 'ReadAsync' method call to use the 'ReadAsync(System.Memory, System.Threading.CancellationToken)' overload.
```
(ideally without the type name on the method).

Reply:
@stephentoub I agree that the message text itself needs to be updated, and to remove the type name on the method. I'll do that soon.

Reply:
Thank you @Youssef1313 for taking care of this. I replied in the other [outdated comment](https://github.com/dotnet/roslyn-analyzers/pull/3677#discussion_r432248458), I understand what the reported problem is, and thanks to both for pointing out my mistake and correcting me.
Stephen described accurately above what the original message was intending: offering the user the signature of the alternative method to use, including the parameter types, and I agree that that's what we should show.

Reply:
@stephentoub

I couldn't get it exactly to be:

> Change the 'ReadAsync' method call to use the 'ReadAsync(System.Memory, System.Threading.CancellationToken)' overload.

Would it be okay if it is:

"Stream.ReadAsync(Memory<byte>, CancellationToken)" instead of
"ReadAsync(System.Memory, System.Threading.CancellationToken)"

Reply:
I think that's be good enough. The user would not lose context if we offer that message, because they can explore the additional overloads and see the full name of those parameter types.